product_sale_manufactured_for/views/sale_order.xml

```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2021 Camptocamp
+     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl) -->
+<odoo>
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="name">sale.order.form</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form" />
+        <field name="arch" type="xml">
+            <xpath
+                expr="//field[@name='order_line']/form//field[@name='product_id']"
+                position="attributes"
+            >
+                <!-- The domain is string, so we copy all of it, and add the new condition. -->
+                <attribute name="domain">
+                    [('sale_ok', '=', True),
```

Comment:
IMHO, this is the weakness of the implementation as it could break inheritance

Reply:
did you recommend some change improve this

Reply:
Nope, because there is no alternative for now.

But, careful, base domain has changed in v18. There is no more company criteria.

See https://github.com/odoo/odoo/blob/18.0/addons/sale/views/sale_order_views.xml#L419

Reply:
So now we still accept this implementation